### DISCUSSION TITLE
Please Change the Background of Internal PDF Viewer Back to Grey!

### DISCUSSION THREAD

**Discussion by xybermindx** [texstudio-org/texstudio#3656] [General]
I switched back to version 4.7.3 because of this. In the 4.8.0, the background is the white one, which looks the same as the document/paper. Please let the grey background back or at least make it as an option. Thanks!

<img width="410" alt="12" src="https://github.com/texstudio-org/texstudio/assets/34178474/1e423d65-fa8d-4786-90c7-354f43571aaa">

**Comment by sunderme** [comment]
the background was not changed. Maybe play around with txs styles ?

**Reply by xybermindx** [reply]
lol. I would like to stay in 4.7.3 for a while and possibly update in the next version....

**Comment by sunderme** [comment]
and maybe https://github.com/texstudio-org/texstudio/issues/3643 ?

**Reply by xybermindx** [reply]
<img width="294" alt="2" src="https://github.com/texstudio-org/texstudio/assets/34178474/3504c2c9-59e2-4a5b-9e80-602e1ff1d088">
Here is what I see on 4.8.0. So distracting....

**Comment by xybermindx** [comment]
Neither of the above replies work. I switched back to 4.7.3 again. Will not upgrade to 4.8.0 again, wait till the next version.

**Comment by octaeder** [comment]
Not reproducible on Win10 with current dev. Never observed such an issue while 4.8.0 was in development.

**Reply by xybermindx** [reply]
I am using Win 11

**Comment by jstover79** [comment]
This issue persists on Windows 11 and TeXstudio version 4.9.1. The interval viewer background is dark grey when using the embedded viewer, but the windowed viewer only shows the dark grey in between pages and on the right edge of the page (just a thin outline).
<img width="1174" height="120" alt="image" src="https://github.com/user-attachments/assets/5317468f-025f-4c94-b790-568afe21161a" />
This is a problem for me as I like to work with the editor on one monitor and the viewer on another. A workaround is to use the embedded viewer, but expand the app fully across both monitors. Annoying, but works in the meantime.

**Reply by nengwanx** [reply]
Same. I have tolerated this issue for years, and now get used to it already.

**Comment by octaeder** [comment]
Worked for me with win10 as mentioned above and is also fine with win11. I use these settings:

<img width="190" height="91" alt="grafik" src="https://github.com/user-attachments/assets/2ae3599e-4140-4a08-b97f-4d22acdaf3e7" />

windowed or embedded makes no difference.

**Comment by octaeder** [comment]
PDFDocument.cpp, l. 772: `setBackgroundRole(QPalette::Base);`

[QPalette::ColorRole](https://doc.qt.io/qt-6/qpalette.html#ColorRole-enum):
> QPalette::Base
> Used mostly as the background color for text entry widgets, but can also be used for other painting - such as the background of combobox drop down lists and toolbar handles. It is usually white or another light color.

Works also with style `windows11` and color scheme `Modern` or `Classic` (in all cases OS Windows 11 in dark mode).

<img width="162" height="91" alt="grafik" src="https://github.com/user-attachments/assets/46b887dc-795e-466e-8fbc-9833fa5f29ec" />

But things get different when changing OS Windows 11 to light mode. For embedded viewer gray is used, so is ok. But for windowed viewer color changes to white (use as above style `windows11` and color scheme `Modern` or `Classic`):

<img width="385" height="225" alt="grafik" src="https://github.com/user-attachments/assets/c7f87513-f6e5-4600-82eb-0432312af325" />

No issue with style `Fusion` and color scheme `Modern` or `Modern - dark`:

<img width="381" height="223" alt="grafik" src="https://github.com/user-attachments/assets/ac2db970-fd5f-4d9e-9d4e-346092344d30" />

Settings:

<img width="265" height="143" alt="grafik" src="https://github.com/user-attachments/assets/90135a5c-f5ea-4c9a-b3ce-382c6e40414c" />


#### Maybe Qt or Windows 11 changed color selection.

Since txs doesn't use an explicit color, I think your best options are as mentioned above: checkout other styles and color schemes.

**Reply by jstover79** [reply]
Thanks so much for this tip! I had never messed with the theme settings. The background color matching the paper colors occurs when the style is set to `windows11` or `default`. All other styles fix that issue. 

I wonder if there is a way to customize that background color in a *.txsprofile file?

Either way, my issue is solved. Thanks!

**Reply by nengwanx** [reply]
Also work for me. Now the problem is finally fixed.

**Comment by octaeder** [comment]
@sunderm I made a quick check with

> QPalette::Window	A general background color.

But I see no effect.

**Comment by octaeder** [comment]
I don't expect that there will be plans to make this configurable. For this themes/styles exist.

**Comment by sunderme** [comment]
fixed
